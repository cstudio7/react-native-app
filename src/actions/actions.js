import {
  FETCH_NAMES_REQUEST,
  LISTSCREEN_NAME_FAVORITE,
  FAVSCREEN_NAME_FAVORITE,
  LISTSCREEN_NAME_UNFAVORITE,
  FAVSCREEN_NAME_UNFAVORITE,
  LISTSCREEN_NAME_OPEN,
  FAVSCREEN_NAME_OPEN,
  LISTSCREEN_VIEW,
  FAVSCREEN_VIEW,
  LISTSCREEN_SCROLL,
  FAVSCREEN_SCROLL
} from '../constants/ActionTypes';

const isScreenFavorites = screen => screen === 'Favorites';

export const favoritesScreenView = () => ({
  type: FAVSCREEN_VIEW
});

export const listScreenView = () => ({
  type: LISTSCREEN_VIEW
});

export const favorite = (screen, payload) => {
  let actionType = '';
  if (isScreenFavorites(screen)) {
    actionType = {
      true: FAVSCREEN_NAME_FAVORITE,
      false: FAVSCREEN_NAME_UNFAVORITE
    }[payload.name.isFavorite];
  } else {
    actionType = {
      true: LISTSCREEN_NAME_FAVORITE,
      false: LISTSCREEN_NAME_UNFAVORITE
    }[payload.name.isFavorite];
  }

  return {
    type: actionType,
    payload
  };
};

export const scrollEvent = screen => {
  let type = LISTSCREEN_SCROLL;
  if (isScreenFavorites(screen)) {
    type = FAVSCREEN_SCROLL;
  }
  return { type };
};

export const openNameScreen = screen => {
  let type = LISTSCREEN_NAME_OPEN;
  if (isScreenFavorites(screen)) {
    type = FAVSCREEN_NAME_OPEN;
  }
  return { type };
};

export const loadNames = () => ({
  type: FETCH_NAMES_REQUEST
})
