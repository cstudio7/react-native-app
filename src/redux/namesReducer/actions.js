import {
  FETCH_NAMES_SUCCESS,
  LISTSCREEN_NAME_FAVORITE,
  FAVSCREEN_NAME_FAVORITE,
  LISTSCREEN_NAME_UNFAVORITE,
  FAVSCREEN_NAME_UNFAVORITE,
  LISTSCREEN_VIEW,
  FAVSCREEN_VIEW,
  LISTSCREEN_SCROLL,
  FAVSCREEN_SCROLL,
  LISTSCREEN_NAME_OPEN,
  FAVSCREEN_NAME_OPEN
} from './constants';

export const fetchNamesSuccess = payload => ({
  type: FETCH_NAMES_SUCCESS,
  payload
});

export const favorite = (screen, payload) => {
  let actionType = '';
  if (screen === 'Favorites') {
    actionType = {
      true: FAVSCREEN_NAME_FAVORITE,
      false: FAVSCREEN_NAME_UNFAVORITE
    }[payload.name.isFavorite];
  } else {
    actionType = {
      true: LISTSCREEN_NAME_FAVORITE,
      false: LISTSCREEN_NAME_UNFAVORITE
    }[payload.name.isFavorite];
  }

  return {
    type: actionType,
    payload
  };
};

export const favoritesScreenView = () => ({
  type: FAVSCREEN_VIEW
});

export const listScreenView = () => ({
  type: LISTSCREEN_VIEW
});

export const favoritesScreenScroll = () => ({
  type: FAVSCREEN_SCROLL
});

export const listScreenScroll = () => ({
  type: LISTSCREEN_SCROLL
});

export const openNameScreen = screen => {
  let type = LISTSCREEN_NAME_OPEN;
  if (screen === 'Favorites') {
    type = FAVSCREEN_NAME_OPEN;
  }
  return { type };
};
