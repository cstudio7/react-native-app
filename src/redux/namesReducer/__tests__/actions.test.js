import { fetchNamesSuccess, favorite } from '../actions';

const mockedName = Object.freeze({ name: 'artur', isFavorite: true });

it('creates a FETCH_NAMES_SUCCESS action', () => {
  const mockedNames = Object.freeze([mockedName]);
  expect(
    fetchNamesSuccess({ gender: 'male', names: mockedNames })
  ).toMatchSnapshot();
});

it('creates a FAVSCREEN_NAME_FAVORITE action', () => {
  const payload = { gender: 'male', name: mockedName };
  const page = 'favorites';
  expect(favorite(payload, page)).toMatchSnapshot();
});

it('creates a LISTSCREEN_NAME_UNFAVORITE action', () => {
  const payload = {
    gender: 'male',
    name: {
      name: 'artur',
      isFavorite: false
    }
  };
  const page = undefined;
  expect(favorite(payload, page)).toMatchSnapshot();
});
