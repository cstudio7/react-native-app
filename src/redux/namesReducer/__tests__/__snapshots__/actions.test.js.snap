// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`creates a FAVSCREEN_NAME_FAVORITE action 1`] = `
Object {
  "payload": Object {
    "gender": "male",
    "name": Object {
      "isFavorite": true,
      "name": "artur",
    },
  },
  "type": "FAVSCREEN_NAME_FAVORITE",
}
`;

exports[`creates a FAVSCREEN_NAME_OPEN action 1`] = `
Object {
  "type": "FAVSCREEN_NAME_OPEN",
}
`;

exports[`creates a FAVSCREEN_SCROLL action 1`] = `
Object {
  "type": "FAVSCREEN_SCROLL",
}
`;

exports[`creates a FAVSCREEN_VIEW action 1`] = `
Object {
  "type": "LISTSCREEN_VIEW",
}
`;

exports[`creates a FETCH_NAMES_SUCCESS action 1`] = `
Object {
  "payload": Object {
    "gender": "male",
    "names": Array [
      Object {
        "isFavorite": true,
        "name": "artur",
      },
    ],
  },
  "type": "FETCH_NAMES_SUCCESS",
}
`;

exports[`creates a LISTSCREEN_NAME_OPEN action 1`] = `
Object {
  "type": "LISTSCREEN_NAME_OPEN",
}
`;

exports[`creates a LISTSCREEN_NAME_UNFAVORITE action 1`] = `
Object {
  "payload": Object {
    "gender": "male",
    "name": Object {
      "isFavorite": false,
      "name": "artur",
    },
  },
  "type": "LISTSCREEN_NAME_UNFAVORITE",
}
`;

exports[`creates a LISTSCREEN_SCROLL action 1`] = `
Object {
  "type": "LISTSCREEN_SCROLL",
}
`;

exports[`creates a LISTSCREEN_VIEW action 1`] = `
Object {
  "type": "FAVSCREEN_VIEW",
}
`;
